pkg/scaling/executor/scale_scaledobjects.go

```diff
@@ -54,54 +54,72 @@ func (e *scaleExecutor) RequestScale(ctx context.Context, scaledObject *kedav1al
 		currentReplicas = currentScale.Spec.Replicas
 	}
 
-	switch {
-	case currentReplicas == 0 && isActive:
-		// current replica count is 0, but there is an active trigger.
-		// scale the ScaleTarget up
-		e.scaleFromZero(ctx, logger, scaledObject, currentScale)
-	case !isActive &&
-		isError &&
-		scaledObject.Spec.Fallback != nil &&
-		scaledObject.Spec.Fallback.Replicas != 0:
-		// there are no active triggers, but a scaler responded with an error
-		// AND
-		// there is a fallback replicas count defined
-
-		// Scale to the fallback replicas count
-		e.doFallbackScaling(ctx, scaledObject, currentScale, logger, currentReplicas)
-
-	case !isActive &&
-		currentReplicas > 0 &&
-		(scaledObject.Spec.MinReplicaCount == nil || *scaledObject.Spec.MinReplicaCount == 0):
-		// there are no active triggers, but the ScaleTarget has replicas.
-		// AND
-		// There is no minimum configured or minimum is set to ZERO. HPA will handles other scale down operations
-
-		// Try to scale it down.
-		e.scaleToZero(ctx, logger, scaledObject, currentScale)
-	case !isActive &&
-		scaledObject.Spec.MinReplicaCount != nil &&
-		currentReplicas < *scaledObject.Spec.MinReplicaCount:
-		// there are no active triggers
-		// AND
-		// ScaleTarget replicas count is less than minimum replica count specified in ScaledObject
-		// Let's set ScaleTarget replicas count to correct value
-		_, err := e.updateScaleOnScaleTarget(ctx, scaledObject, currentScale, *scaledObject.Spec.MinReplicaCount)
-		if err == nil {
-			logger.Info("Successfully set ScaleTarget replicas count to ScaledObject minReplicaCount",
-				"Original Replicas Count", currentReplicas,
-				"New Replicas Count", *scaledObject.Spec.MinReplicaCount)
+	if isActive {
+		switch {
+		case scaledObject.Spec.IdleReplicaCount != nil && currentReplicas < *scaledObject.Spec.MinReplicaCount,
```

Comment:
`minReplicaCount` could be nil, right?

Reply:
This should cover this, but you are right, I should check that here as well. Thx!
https://github.com/kedacore/keda/pull/1958/files#diff-1492a6dd820b37e93cd04e4f0fd2f077e1fde7056f91c576e042a58e0305df27R326

Reply:
Fixed in https://github.com/kedacore/keda/pull/1958/commits/45139fb1661874f4eb911b700d881aecb61cae3f